pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://mesolite.cluster.engflow.com/invocations/default/c9ee2f1c-6629-40da-9461-4700c4596e00?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3Rlc3RjY2wvd29ya2xvYWQvc2NoZW1hY2hhbmdlOnNjaGVtYWNoYW5nZV90ZXN0) on master @ [4b14e665949a064e0319dcbce157cdbe0b232758](https://github.com/cockroachdb/cockroach/commits/4b14e665949a064e0319dcbce157cdbe0b232758):


```
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	pkg/workload/schemachange/schemachange.go:513
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	pkg/workload/schemachange/schemachange.go:610
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func5.func7
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:154
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	Wraps: (4) ***UNEXPECTED ERROR; Failed to generate a random operation
        	            	Wraps: (5) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).randOp
        	            	  | 	pkg/workload/schemachange/operation_generator.go:219
        	            	  | [...repeated from below...]
        	            	Wraps: (6) failed generating operation: selectStmt
        	            	Wraps: (7) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).selectStmt
        	            	  | 	pkg/workload/schemachange/operation_generator.go:4669
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).randOp
        	            	  | 	pkg/workload/schemachange/operation_generator.go:206
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	pkg/workload/schemachange/schemachange.go:500
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	pkg/workload/schemachange/schemachange.go:610
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func5.func7
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:154
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1700
        	            	Wraps: (8) error getting random table name
        	            	Wraps: (9) ERROR: column "crdb_internal_column_1_name_placeholder" does not exist, referenced in "crdb_internal_column_1_name_placeholder IS NOT NULL" (SQLSTATE 42703)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *withstack.withStack (6) *errutil.withPrefix (7) *withstack.withStack (8) *errutil.withPrefix (9) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 106
Total Schema Statements Succeeded = 106
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 87
Total Transactions Rolled Back = 43
Total Transactions Executed = 130
    schema_change_external_test.go:118: backup, tracing data, and system table dumps in /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/testlogs/pkg/ccl/testccl/workload/schemachange/schemachange_test/test.outputs/logTestWorkload1042295545
    schema_change_external_test.go:127: running DROP with use_declarative_schema_changer = on
    panic.go:635: -- test log scope end --
test logs left over in: outputs.zip/logTestWorkload1042295545
--- FAIL: TestWorkload (51.34s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #144033 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [violates foreign key constraint] [C-test-failure O-robot P-2 T-sql-foundations branch-release-25.1]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19862767?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19862767?buildTab=artifacts#/) on master @ [f8f14d27ef5de3a41e0e81e155a1188a457210b2](https://github.com/cockroachdb/cockroach/commits/f8f14d27ef5de3a41e0e81e155a1188a457210b2):


```
    "queryArgs": [
     "table_w0_10",
     "public"
    ],
    "result": false
   }
  ],
  "previousStatements": [
   "ALTER TABLE public.table_w0_10 RENAME TO public.table_w1_69"
  ]
 }
}
    schema_change_external_test.go:165: 
        	Error Trace:	pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:165
        	Error:      	Received unexpected error:
        	            	***UNEXPECTED ERROR; Received an unexpected execution error.: ERROR: cannot rename relation "public.table_w0_10" because trigger "trigger_w1_66" on table "table_w0_10" depends on it (SQLSTATE 2BP01)
        	            	(1) forced error mark
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*opStmt).executeStmt
        	            	  | 	pkg/workload/schemachange/operation_generator.go:3322
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	pkg/workload/schemachange/schemachange.go:523
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	pkg/workload/schemachange/schemachange.go:611
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func5.func7
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:154
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_arm64.s:1223
        	            	Wraps: (4) ***UNEXPECTED ERROR; Received an unexpected execution error.
        	            	Wraps: (5) ERROR: cannot rename relation "public.table_w0_10" because trigger "trigger_w1_66" on table "table_w0_10" depends on it (SQLSTATE 2BP01)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 53
Total Schema Statements Succeeded = 53
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 38
Total Transactions Rolled Back = 28
Total Transactions Executed = 66
    schema_change_external_test.go:118: backup, tracing data, and system table dumps in /artifacts/tmp/_tmp/67537770b84c3a4ea177539c1e46bd43/logTestWorkload2072578434
    schema_change_external_test.go:127: running DROP with use_declarative_schema_changer = off
    panic.go:635: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/67537770b84c3a4ea177539c1e46bd43/logTestWorkload2072578434
--- FAIL: TestWorkload (33.55s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #144033 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [violates foreign key constraint] [C-test-failure O-robot P-2 T-sql-foundations branch-release-25.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19872395?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19872395?buildTab=artifacts#/) on master @ [9e60b94887bd600cea9cdea0b705948879b8ceef](https://github.com/cockroachdb/cockroach/commits/9e60b94887bd600cea9cdea0b705948879b8ceef):


```
  }
 ],
 "expectedExecErrors": "23502",
 "expectedCommitErrors": "",
 "message": "ROLLBACK; Successfully got expected execution error.: ERROR: null value in column \"col_w0_81_w1_403\" violates not-null constraint (SQLSTATE 23502)"
}
{
 "workerId": 0,
 "clientTimestamp": "16:09:13.922909",
 "ops": [
  "BEGIN",
  {
   "sql": "CREATE TABLE IF NOT EXISTS schema_w0_125.table_w0_206 (col206_w0_207 INT2 NOT NULL, col206_w0_208 FLOAT8 NULL, col206_w0_209 VARCHAR NOT NULL, col206_w0_210 FLOAT8 NOT NULL, col206_w0_211 TIMETZ NULL, col206_w0_212 FLOAT8 NOT NULL, col206_w0_213 PG_LSN, col206_w0_214 BIT(9) NULL, col206_w0_215 INT2, col206_w0_216 INT8 AS (col206_w0_207 + col206_w0_215) VIRTUAL, col206_w0_217 FLOAT8 NULL AS (abs(col206_w0_208)) VIRTUAL, col206_w0_218 FLOAT8 NULL AS ((col206_w0_212 + col206_w0_210) + col206_w0_208) VIRTUAL, col206_w0_219 FLOAT8 NULL AS (col206_w0_208 + col206_w0_210) VIRTUAL, col206_w0_220 FLOAT8 NOT NULL AS (abs(col206_w0_212)) VIRTUAL, col206_w0_221 FLOAT8 NOT NULL AS (abs(col206_w0_212)) VIRTUAL, col206_w0_222 FLOAT8 NULL AS (col206_w0_208 + col206_w0_212) STORED, PRIMARY KEY (col206_w0_210 DESC), UNIQUE (col206_w0_221 ASC, col206_w0_208 ASC) STORING (col206_w0_209, col206_w0_214, col206_w0_222), FAMILY (col206_w0_210, col206_w0_207, col206_w0_208, col206_w0_215, col206_w0_211, col206_w0_222, col206_w0_214, col206_w0_212), FAMILY (col206_w0_213), FAMILY (col206_w0_209))"
  },
  {
   "sql": "CREATE TABLE IF NOT EXISTS schema_w0_202.table_w0_223 (col223_w0_224 REFCURSOR NOT NULL, col223_w0_225 STRING COLLATE da_DK NOT NULL, INDEX (lower(CAST(col223_w0_225 AS STRING)) ASC) STORING (col223_w0_224), INDEX (lower(CAST(col223_w0_224 AS STRING)) DESC), INDEX (lower(CAST(col223_w0_225 AS STRING)) DESC, col223_w0_225 DESC), UNIQUE (col223_w0_225), FAMILY (col223_w0_224, col223_w0_225)) WITH ('schema_locked' = true)"
  },
  "COMMIT"
 ],
 "expectedExecErrors": "",
 "expectedCommitErrors": "",
 "message": ""
}
{
 "workerId": 1,
 "clientTimestamp": "16:09:14.336702",
 "ops": [
  "BEGIN",
  {
   "sql": "INSERT INTO public.table_w1_129 (col129_w1_130, col129_w1_131, col129_w1_132, col129_w1_133, col129_w1_134, col129_w1_135, col129_w1_136, col129_w1_137) VALUES ('01040000A0E6100000050000000101000080648AE8784F75184031D25880E3A53FC085B71A15A453FE410101000080F228F503A42D48409A8134A0C3FC4B40D6502E024060AAC1010100008078D480D9D2CF61407055D91169AE52C02C0EFAA3604BEFC10101000080382324C254404BC034DBB7D0BA864A40484DF4D044C6F9C101010000800953884FAD645DC084BCD1175BFD1A40C8AC738AB4EEE841':::GEOGRAPHY,1082:::OID,e'q0^\\\\\\x14?v'::STRING,1.361311596484003:::FLOAT8,3802:::OID,(-80):::INT8,'235.137.128.48/26':::INET,'''eoWKhsK'' \u0026 ''LRry''':::TSQUERY)",
   "potentialExecErr": "23502,23505,23514,42501"
  },
  "COMMIT"
 ],
 "expectedExecErrors": "",
 "expectedCommitErrors": "",
 "message": ""
}
Schema Workload Stats
Total Schema Statements Executed = 152
Total Schema Statements Succeeded = 152
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 93
Total Transactions Rolled Back = 107
Total Transactions Executed = 200
    schema_change_external_test.go:127: running DROP with use_declarative_schema_changer = off
    schema_change_external_test.go:129: error executing query="DROP DATABASE schemachange CASCADE" args=[]: pq: table "view_w1_402" is already being dropped
    panic.go:635: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/67537770b84c3a4ea177539c1e46bd43/logTestWorkload4040618111
--- FAIL: TestWorkload (346.74s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #144033 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [violates foreign key constraint] [C-test-failure O-robot P-2 T-sql-foundations branch-release-25.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/workload/schemachange/optype.go
+++ b/pkg/workload/schemachange/optype.go
@@ -315,7 +315,7 @@ var opWeights = []int{
 	dropSchema:                        1,
 	dropSequence:                      1,
 	dropTable:                         1,
-	dropTrigger:                       1,
+	dropTrigger:                       0, // TODO(147981): re-enable once we fix orphaning of backref in DROP TRIGGER
 	dropView:                          1,
 	renameIndex:                       1,
 	renameSequence:                    1,

